Tactics/Reflect/AxEffects.lean

```diff
@@ -246,7 +279,10 @@ validation will still pass if types are different to those we claim in the
 docstrings
 -/
 def MemoryEffects.validate (eff : MemoryEffects) : MetaM Unit := do
-  sorry
+  let msg := "validating that the axiomatic effects are well-formed"
+  withTraceNode' msg do 
+    eff.traceCurrentState
+    assertHasType eff.effects mkMemory
```

Comment:
You need to run `check` also; `assertHasType` will run `inferType`, and inferType intentionallty does not catch all mallformed expressions!